consensus/task.go

```diff
@@ -0,0 +1,31 @@
+package consensus
+
+import "sync"
+
+type TaskRunner struct {
+	wg     sync.WaitGroup
+	taskCh chan func()
+}
+
+func StartTaskRunner(buf int) *TaskRunner {
+	tr := &TaskRunner{
+		taskCh: make(chan func(), buf),
+	}
+	tr.wg.Add(1)
+	go func() {
+		defer tr.wg.Done()
+		for f := range tr.taskCh {
+			f()
```

Comment:
so we just avoid blocking with single go routine instead of concurrent right?

Reply:
if we have multiple `fireEvents` goroutine running concurrently, subscriber will get overlapped messages, so better to run a single goroutine concurrently with block execution, preserve the execution order of the tasks.